webapp/app/utils.py

```diff
@@ -109,3 +117,216 @@ def entity_to_int(s):
+    """
+    output: Dict[str, Any] = {  # ignore cluster columns
+        k: getattr(addr, k) for k in table_cols if 'cluster' not in k
+    }
+    output['conf'] = round(output['conf'], 3)
+    del output['meta_data']  # Q: should we keep this?
+
+    for k in to_remove:
+        if k in output:
+            del output[k]
+    for key, func in to_transform:
+        output[key] = func(output[key])
+    return output
+
+
+def default_response() -> Dict[str, Any]:
+    output: Dict[str, Any] = {
+        'data': {
+            'query': {
+                'address': '', 
+                'metadata': {}, 
+                'tornado': {
+                    'exact_match': {},
+                    'gas_price': {},
+                }
+            },
+            'cluster': [],
+            'metadata': {
+                'cluster_size': 0,
+                'num_pages': 0,
+                'page': 0,
+                'limit': 50,
+                'filter_by': {
+                    'min_conf': 0,
+                    'max_conf': 1,
+                    'entity': '*',
+                    'name': '*',
+                },
+                'schema': {
+                    ADDRESS_COL: {
+                        'type': 'string',
+                    },
+                    CONF_COL: {
+                        'type': 'float',
+                        'values': [0, 1],
+                    },
+                    ENTITY_COL: {
+                        'type': 'category',
+                        'values': [
+                            EOA, DEPOSIT, EXCHANGE, DEX, DEFI, ICO_WALLET, MINING],
+                    },
+                    NAME_COL: {
+                        'type': 'string',
+                    },
+                },
+                'sort_default': {
+                    'attribute': ENTITY_COL,
+                    'descending': False
+                }
+            }
+        },
+        'success': 0,
+    }
+    return output
+
+
+class RequestChecker:
+    """
+    Given a request object with its args, make sure that it is 
+    providing valid arguments.
+    """
+    
+    def __init__(
+        self,
+        request: Any,
+        table_cols: List[str],
+        entity_key: str = 'entity',
+        conf_key: str = 'confidence',
+        name_key: str = 'name',
+        default_page: int = 0,
+        default_limit: int = 50,
+    ):
+        self._request: Any = request
+        self._table_cols: List[str] = table_cols
+        self._entity_key: str = entity_key
+        self._conf_key: str = conf_key
+        self._name_key: str = name_key
+        self._default_page: int = default_page
+        self._default_limit: int = default_limit
+
+        self._params: Dict[str, Any] = {}
+
+    def check(self):
+        return (self._check_address() and
+                self._check_page() and
+                self._check_sort_by() and 
+                self._check_filter_by())
+
+    def _check_address(self) -> bool:
```

Comment:
awesome!!! thx for doing this. I'll take out the checks from the front-end code then.